apis/apiextensions/v1/xrd_validation.go

```diff
@@ -0,0 +1,54 @@
+package v1
+
+import (
+	"fmt"
+
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// Validate checks that the supplied CompositeResourceDefinition spec is logically valid.
+func (c *CompositeResourceDefinition) Validate() (warns []string, errs field.ErrorList) {
+	type validationFunc func() field.ErrorList
+	validations := []validationFunc{
+		c.validateConversion,
+	}
+	for _, f := range validations {
+		errs = append(errs, f()...)
+	}
+	return nil, errs
+}
+
+// validateConversion checks that the supplied CompositeResourceDefinition spec
```

Comment:
seems like the end of the sentance is missing

Reply:
fixed 👍